File: solutions/go/pascals-triangle/2/pascals_triangle.go
package pascal

func Triangle(n int) (res [][]int) {
	if n > 0 {
    	row := []int{1}
    	res = [][]int{row}
    	for i := 1; i < n; i++ {
        	next := append(append([]int{}, row...), 0)
        	for j := 0; j < len(row); j++ {
            	next[j + 1] += row[j]
	        }
    	    res, row = append(res, next), next
    	}
    }
    return
}

File: solutions/clojure/pascals-triangle/1/src/pascals_triangle.clj
(ns pascals-triangle)

(def triangle
  (iterate (fn [row]
             (let [prev1 (conj row (bigint 0))
                   prev2 (into [(bigint 0)] row)]
               (mapv + prev1 prev2)))
           [(bigint 1)]))

(defn row [n]
  (last (take n triangle)))
